src/modules/dashboard/sagas/actions/createEditDomainMotion.ts

```diff
@@ -0,0 +1,201 @@
+
+import { ContextModule, TEMP_getContext } from '~context/index';
+import { Action, ActionTypes, AllActions } from '~redux/index';
+import { putError, takeFrom, routeRedirect } from '~utils/saga/effects';
+import {
+  createTransaction,
+  createTransactionChannels,
+  getTxChannel,
+} from '../../../core/sagas';
+import { ipfsUpload } from '../../../core/sagas/ipfs';
+import {
+  transactionReady,
+  transactionPending,
+  transactionAddParams,
+} from '../../../core/actionCreators';
+
+function* createEditDomainMotion({
+  payload: {
+    colonyAddress,
+    colonyName,
+    domainName,
+    domainColor,
+    domainPurpose,
+    annotationMessage,
+    domainId: editDomainId,
+    isCreateDomain,
+    parentId = ROOT_DOMAIN_ID,
+  },
+  meta: { id: metaId, history },
+  meta,
+}: Action<ActionTypes.COLONY_MOTION_DOMAIN_CREATE_EDIT>) {
+  let txChannel;
+  try {
+    /*
+     * Validate the required values
+     */
+    if (!domainName) {
+      throw new Error('A domain name is required to create a new domain');
+    }
+
+    if (!isCreateDomain && !editDomainId) {
+      throw new Error('A domain id is required to edit domain');
+    }
+    /* additional editDomain check is for the TS to not ring alarm in getPermissionProofs */
+    const domainId = !isCreateDomain && editDomainId ? editDomainId : parentId;
+
+    const context = TEMP_getContext(ContextModule.ColonyManager);
+    const colonyClient = yield context.getClient(
+      ClientType.ColonyClient,
+      colonyAddress,
+    );
+
+    const [permissionDomainId, childSkillIndex] = yield call(
+      getPermissionProofs,
+      colonyClient,
+      domainId,
+      ColonyRole.Architecture,
+    );
+
+    const { skillId } = yield call(
+      [colonyClient, colonyClient.getDomain],
+      ROOT_DOMAIN_ID,
+    );
+
+    const { key, value, branchMask, siblings } = yield call(
+      colonyClient.getReputation,
+      skillId,
+      AddressZero,
+    );
+
+    txChannel = yield call(getTxChannel, metaId);
+
+    // setup batch ids and channels
+    const batchKey = 'createRootMotion';
+
+    const {
+      createRootMotion,
+      annotateMotion,
+    } = yield createTransactionChannels(metaId, [
+      'createRootMotion',
+      'annotateMotion',
+    ]);
+
+    /*
+     * Upload domain metadata to IPFS
+     */
+    let domainMetadataIpfsHash = null;
+    domainMetadataIpfsHash = yield call(
+      ipfsUpload,
+      JSON.stringify({
+        domainName,
+        domainColor,
+        domainPurpose,
+      }),
+    );
+
+    const encodedAction = colonyClient.interface.functions[
+      isCreateDomain
+        ? 'addDomain(uint256,uint256,uint256,string)'
+        : 'editDomain'
+    ].encode([
+      permissionDomainId,
+      childSkillIndex,
+      domainId,
+      domainMetadataIpfsHash,
+    ]);
+
+    // create transactions
+    yield fork(createTransaction, createRootMotion.id, {
+      context: ClientType.VotingReputationClient,
+      methodName: 'createRootMotion',
```

Comment:
This is regarding all the mentioned motions (createEditDomain, moveFunds, createPayment that happen with root domain), I have talked to Alex and he said that it theoretically can be done via both rootMotion & domainMotion but rootMotion requires less gas so it should be a preferred option. He is even thinking of introducing an error if root domain is involved but the method called is `createDomainMotion`.

"After talking with some of the frontend team, there is an ambiguity here for making motions for actions that require a non-root permission in the root domain. They can be made through both createRootMotion and createDomainMotion. Do you think we should remove that ambiguity? I've told the fronted team to, if a motion is being created in root, they should use createRootMotion as it will be cheaper, gas wise - so should we error if createDomainMotion is tried to be used to create a motion in root?"